src/main/scala/iog/psg/cardano/CardanoApiCodec.scala

```diff
@@ -445,12 +446,17 @@ object CardanoApiCodec {
         case e: Exception => errorUnparseableResult(e)
       }
 
-    private def sequenceCardanoApiResponses[T](resp: Seq[CardanoApiResponse[T]]): CardanoApiResponse[Seq[T]] =
-      resp.find(_.isLeft) match {
-        case Some(Left(error)) => Left(error)
-        case None => Right(resp.flatMap(_.toOption))
+    private def sequenceCardanoApiResponses[T](responses: Seq[CardanoApiResponse[T]]): CardanoApiResponse[List[T]] = {
+      @tailrec
```

Comment:
Why couldn't you use the foldLeft? It's shorter, it's standard, it's efficient in the optimistic case, it doesn't require a conversion toList....?

Reply:
I could but this way, its faster, because foldLeft if will encounter error lets say on 2nd record, wont break on it, will continune processing until end of the list, but doing tail recursive method it will break on first error, so in case of errors its very fast :) thats why I decided to go for it

Reply:
Or possibly 'reduce'?

Reply:
Use `foldLeft` or `reduce` if it works please.